osu.Game/Screens/Play/Player.cs

```diff
@@ -541,10 +541,8 @@ protected void PerformExit(bool showDialogFirst)
                 }
 
                 // if the score is ready for display but results screen has not been pushed yet (e.g. storyboard is still playing beyond gameplay), then transition to results screen instead of exiting.
-                if (prepareScoreForDisplayTask != null)
+                if (prepareScoreForDisplayTask != null && completionProgressDelegate == null)
```

Comment:
I didn't clear `completionProgressDelegate` after it's cancelled at the beginning of the function because that would make the score display on exit.